From 4485bd39b06e77059be370231a966545b6141f19 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 16 May 2025 15:40:59 +0800
Subject: [PATCH] support cross compiling

Use target directory to instead of native directory

The cuda_toolchain.cmake in meta-tegra defines var-CUDA_TOOLKIT_TARGET_DIR
for cross compiling

Upstream-Status: Inappropriate [Yocto specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/import/CMakeLists.txt b/src/import/CMakeLists.txt
index 5343d877..6b22e16f 100644
--- a/src/import/CMakeLists.txt
+++ b/src/import/CMakeLists.txt
@@ -79,7 +79,7 @@ if(CMAKE_CUDA_COMPILER)
     set(OLLAMA_CUDA_INSTALL_DIR ${OLLAMA_INSTALL_DIR}/cuda_v${CUDAToolkit_VERSION_MAJOR})
     install(TARGETS ggml-cuda
         RUNTIME_DEPENDENCIES
-            DIRECTORIES ${CUDAToolkit_BIN_DIR} ${CUDAToolkit_LIBRARY_DIR}
+            DIRECTORIES ${CUDA_TOOLKIT_TARGET_DIR}/bin ${CUDA_TOOLKIT_TARGET_DIR}/lib
             PRE_INCLUDE_REGEXES cublas cublasLt cudart
             PRE_EXCLUDE_REGEXES ".*"
         RUNTIME DESTINATION ${OLLAMA_CUDA_INSTALL_DIR} COMPONENT CUDA
-- 
2.34.1

